include/onnxruntime/core/session/onnxruntime_cxx_api.h

```diff
@@ -3284,5 +3286,64 @@ struct Model : detail::ModelImpl<OrtModel> {
   explicit Model(const std::vector<DomainOpsetPair>& opsets);
 #endif
 };
+
+namespace detail {
+template <typename T>
+struct ConstKernelDefImpl : Base<T> {
+  using B = Base<T>;
+  using B::B;
+
+  ///< Wraps OrtEpApi::KernelDef_GetOperatorType
+  const char* GetOperatorType() const;
+
+  ///< Wraps OrtEpApi::KernelDef_GetDomain
+  const char* GetDomain() const;
+
+  ///< Wraps OrtEpApi::KernelDef_GetSinceVersion
+  std::pair<int, int> GetSinceVersion() const;
+
+  ///< Wraps OrtEpApi::KernelDef_GetExecutionProvider
+  const char* GetExecutionProvider() const;
```

Comment:
If any of the information for any getters is optional, suggest returning a status

Reply:
The underlying C API function `OrtEpApi::KernelDef_GetExecutionProvider` returns the `const char*` directly (doesn't return a status). 

Sorry, I don't fully understand the comment. Is the request to make this return a status instead? What is meant by "information for any getters is optional"?

Reply:
This was a general comment not to be bound to throwing exceptions by default. In this case, the data is returned without error reporting.